nansat/vrt.py

```diff
@@ -918,22 +918,28 @@ def prepare_export_netcdf(self, options, bottomup):
         return options, add_gcps
 
     def copy(self):
-        """Create and return a full copy of a VRT instance"""
+        """Create and return a full copy of a VRT instance with new filenames"""
         if self.dataset.RasterCount == 0:
             vrt = VRT.from_gdal_dataset(self.dataset, metadata=self.dataset.GetMetadata())
         else:
             vrt = VRT.copy_dataset(self.dataset, metadata=self.dataset.GetMetadata())
-
-        vrt.band_vrts = dict(self.band_vrts)
+            replace_filenames = [(self.filename, vrt.filename)]
+            # recursive copy of vrt.vrt
+            if self.vrt is not None:
+                vrt.vrt = self.vrt.copy()
+                replace_filenames.append((self.vrt.filename, vrt.vrt.filename))
+
+            # change reference from original filenames to the new ones
+            for replace_filename in replace_filenames:
```

Comment:
Correct me if I wrong, but if I get that correctly than VRT names actually do not contain any useful information, i.e. it's kinda random generated names? Therefore to fix the bug you are using a substitution of some machine generated (not human readable) names by another also not really representative names. It seems like a mess for me

Reply:
Any constructive comments?

Reply:
Yes.  For instance, could you make vrt filenames more representative?

Reply:
Sure. But what do you mean `more representative`? I honestly don't understand.

Reply:
Ok, sorry, I have an example.
When I open a file with nansat I will have a parameter SourceFilename: /vsimem/ESHZ0FQASS.vrtwhich really does not say me anything. Moreover, different bands in one file could contain different VRTs.
In addition, now you make a substitution of a filename ESHZ0FQASS.vrt by the filename TKR8HPRJY4.vrt which does not help you to trace back any changes or possible bugs. What I would like to see is something like: band_name.vrt or file_copy.vrt or something like that.
Is that possible to do?

Reply:
It is quite challenging to give meaningfull filenames to VRTs.
VRT object doesn't know the name of the band it belongs to. One band can have several VRTs (in case of pixel function). Several bands can refer to one VRT. The VRT that a band refers to can be in band_vrts or in vrt.vrt. So there will never be a simple rule to give a filename for VRT. And to develop complicated rules of naming files? No, thanks!
In fact, VRT filenames are not intended to be seen by users. Think about them as about actual address in memory. Do you know the exact value of each pointer? Do you want them to be decimal? Do you want them to have nice names? I guess not...

However, there is always possibility to identify which VRT is used where. Check the content of the vrt.xml and you will see names of VRTs for each band. Check values vrt.band_vrts and simply print(n.vrt) and you will see the names of vrt itself, sub-VRT and so on.